### DISCUSSION TITLE
Official Meteor NPM SDK

### DISCUSSION THREAD

**Discussion by jankapunkt** [meteor/meteor#12968] [Feature Requests]
This might be a crucial step in making Meteor becoming more integrated with the broader JS ecosystem.

## Background

External clients can't connect to Meteor out-of-the box. Even if they implement the DDP protocol over Websockets, they still miss crucial functionality in order to support

- Pub/sub
- ObserveChanges
- Tracker and reactivity
- Accounts integration
- what else?

An official Meteor SDK might be a perfect solution here

## What already exists?

- [SimpleDDP](https://github.com/Gregivy/simpleddp) is the de-facto implementation
- However not maintained since years
- maybe not feature complete, see the above
- [Meteor-React-Native](https://github.com/meteorrn/meteor-react-native) basically implements all the above (copied the code way back then from the Meteor repo)
- it also maintains an own [Minimongo](https://github.com/meteorrn/minimongo-cache) fork
- Lots of it's code could be removed if an official SDK would be there
- @Grubba27 already created a [meteor-sdk package](https://github.com/Grubba27/meteor-sdk), which is simply a fork of simpleDDP in TypeScript

## Why under the meteor org?

- "official" status, which might be important for developers from other communities to recognize it as the tool to use when connecting their software
- close to the core packages
- bundling development efforts

**Comment by StorytellerCZ** [comment]
One more thing that I would like to see added to the Meteor SDK would be connection to the Meteor Accounts. So a way to call for login/signup to the Meteor server and maintaining the login in a way that if you are on the same domain (or something that makes sense) you don't have to login again if you go to the Meteor app (ie. going from the landing page to the app).
